### PR TITLE
feat(insights): adds create alert buttons to db and queue modules

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds Create Alert menu items to Queue and Database insights module charts that takes users to a prepopulate Create Alert Rule page
image

### CODE REVIEW

File: static/app/views/insights/common/components/chartPanel.tsx [hunk 1]
```diff
@@ -1,21 +1,74 @@
...
+          : undefined;
+      const name = alertConfig.name ?? 'Create Alert';
+      const disabled = !singleProject;
+      return {
+        key: name,
+        label: name,
+        to: alertsUrl,
+        disabled,
+        tooltip: disabled
+          ? t('Alerts are only available for a single project')
```
Comment (DominikB2014) [inline static/app/views/insights/common/components/chartPanel.tsx:65]: Should we also mention what they should do to enable alerts? I.e "please select only one project", or do we think this message is enough to imply that?

File: static/app/views/insights/common/utils/getAlertsUrl.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  query?: string;
+}) {
+  const queryParams = {
+    aggregate: aggregate,
+    dataset: Dataset.GENERIC_METRICS,
+    project: project.slug,
+    eventTypes: 'transaction',
+    query,
+  };
+  return `/alerts/new/metric/?${qs.stringify(queryParams)}`;
```
Comment (DominikB2014) [inline static/app/views/insights/common/utils/getAlertsUrl.tsx:22]: Should we be using `normalizeUrl` here?

**Review by DominikB2014 — APPROVED** [review]
code lgtm to me overall!
Comment on the screenshot: in the context menu, is a entry per type of alert (create published alert and create processed alert) the longterm solution here? I feel like if there is a busy chart, there may be a lot of "create alert" options. Perhaps a single "create alert" button with a modal might be a more consistent and scalable approach in the future.

File: static/app/views/insights/common/components/chartPanel.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/views/insights/common/components/chartPanel.tsx]: Updated the tooltip message!

**Comment by edwardgou-sentry** [comment]
> code lgtm to me overall! Comment on the screenshot: in the context menu, is a entry per type of alert (create published alert and create processed alert) the longterm solution here? I feel like if there is a busy chart, there may be a lot of "create alert" options. Perhaps a single "create alert" button with a modal might be a more consistent and scalable approach in the future.

Yeah the menu dropdown can get lengthy if there's a lot of series. I'm going to leave this for now since it was the easiest way to implement and we can get some design feedback from the team on how we want to change this

**Pull request merged**